crates/bitcoin-da/src/verifier.rs

```diff
@@ -82,29 +80,26 @@ impl DaVerifier for BitcoinVerifier {
         // create hash set of blobs
         let mut blobs_iter = blobs.iter();
 
-        let mut inclusion_iter = inclusion_proof.wtxids.iter();
-
         let prefix = match namespace {
             DaNamespace::ToBatchProver => self.to_batch_proof_prefix.as_slice(),
             DaNamespace::ToLightClientProver => self.to_light_client_prefix.as_slice(),
         };
-        // Check starting bytes tx that parsed correctly is in blobs
-        let mut completeness_tx_hashes = BTreeSet::new();
-
-        for tx in completeness_proof.iter() {
-            let wtxid = tx.compute_wtxid();
-            // make sure it starts with the correct prefix
-            if !wtxid.as_byte_array().starts_with(prefix) {
-                return Err(ValidationError::NonRelevantTxInProof);
+
+        let mut completeness_iter = completeness_proof.iter();
+        for inclusion_wtxid in inclusion_proof.wtxids.iter() {
+            // skip nonrelevant transactions
+            if !inclusion_wtxid.starts_with(prefix) {
+                continue;
             }
 
-            // make sure completeness txs are ordered same in inclusion proof
-            // this logic always start seaching from the last found index
-            // ordering should be preserved naturally
-            let is_found_in_block =
-                inclusion_iter.any(|wtxid_inc| wtxid_inc == wtxid.as_byte_array());
-            if !is_found_in_block {
-                return Err(ValidationError::RelevantTxNotFoundInBlock);
+            // completeness proof tx for the relevant tx must exist
+            let Some(tx) = completeness_iter.next() else {
```

Comment:
https://docs.rs/itertools/0.8.0/itertools/trait.Itertools.html#method.zip_longest

Reply:
with zip_longest we would have to check if the values are some, imo panicing with zip_eq is better. As there is no recovery we can perform if we gave the wrong input.